spec/src/main/asciidoc/overview.asciidoc

```diff
@@ -79,11 +79,60 @@ Instances of `ManagedExecutor` and `ThreadContext` can be injected into CDI bean
     unmanagedCompletionStage.thenApply(threadContext.contextualFunction(function3));
 ----
 
-The container provides default instances of `ManagedExecutor` and `ThreadContext`, as injected above, which are shared within an application with all other injection points for `ManagedExecutor` and `ThreadContext` that do not specify a qualifier. Their configuration is equivalent to `ManagedExecutor.builder().build()` and `ThreadContext.builder().build()`.
+The container provides default instances of `ManagedExecutor` and `ThreadContext`, as injected above, which are shared within an application with all other injection points for `ManagedExecutor` and `ThreadContext` that are neither annotated as named instances nor with a config annotation. Their configuration is equivalent to the default attribute values of `@ManagedExecutorConfig()` and `@ThreadContextConfig()`.
 
 === Injection of Configured Instances
 
-TODO write this section after discussion about the Config annotations is resolved.
+Instances of `ManagedExecutor` and `ThreadContext` with configuration can also be injected into CDI beans via the `@Inject` annotation. The `@ManagedExecutorConfig` and `@ThreadContextConfig` annotations supply the configuration. For example:
+
+[source, java]
+----
+    @Inject @ManagedExecutorConfig(maxAsync = 5, propagated = ThreadContext.SECURITY)
+    ManagedExecutor executor;
+    ...
+    CompletableFuture<Long> stage = executor
+        .newIncompleteFuture()
+        .thenApply(function)
+        .thenAccept(consumer);
+    stage.completeAsync(supplier);
+----
+
+=== Sharing Configured Instances
+
+Configured instances of `ManagedExecutor` and `ThreadContext` can be assigned a `@NamedInstance` qualifier, which can then be used to inject the same instance into other CDI injection points within the application. For example:
+
+[source, java]
+----
+    @Inject @NamedInstance("exec1")
+    @ManagedExecutorConfig(propagated = { ThreadContext.SECURITY, ThreadContext.APPLICATION })
+    ManagedExecutor executor1;
+    ...
+    @Inject
+    void setCompletableFuture(@NamedInstance("exec1") ManagedExecutor exec) {
+        completableFuture = exec.newIncompleteFuture();
+    }
+----
+
+=== Injection of Custom Built Instances
```

Comment:
Although the producer approach is generally obvious to CDI users, I provided this example for a couple of reasons that are specific to ManagedExecutors.
First, I wanted to emphasize that if you build your own managed executor instance for a CDI producer, don't forget to shut it down.
Second, to encourage the use of ApplicationScoped rather than other scopes.
I don't believe the user should need to annotate this with `@Alternative` or `@Priority` at all for this example because it uses the `@NamedInstance` qualifier.  They only matching injection points for this producer will be the ones annotated with `@NamedInstance("exec2")` as shown in the example injection point corresponding to the producer.
This was intended to be a good example of how the user can properly take advantage of CDI producer in a way that fits naturally with the MP Concurrency spec.